checkov/secrets/scan_git_history.py

```diff
@@ -28,134 +30,149 @@ class EnrichedPotentialSecret(TypedDict):
+        self.secrets_by_file_value_type: Dict[str, List[EnrichedPotentialSecret]] = {}
+
+    def set_secret_map(self, file_results: List[PotentialSecret], file_name: str, commit_hash: str) -> None:
+        # First find if secret was moved in the file
+        equal_secret_in_commit: Dict[str, List[str]] = defaultdict(list)
+        for secret in file_results:
+            secret_key = f'{file_name}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
+            equal_secret_in_commit[secret_key].append('added' if secret.is_added else 'removed')
+
+        for secret in file_results:
+            secret_key = f'{file_name}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
+            if all(value in equal_secret_in_commit[secret_key] for value in ['added', 'removed']):
+                continue
+            if secret.is_added:
+                self._add_new_secret(secret_key, commit_hash, secret)
+            if secret.is_removed:
+                self._update_removed_secret(secret_key, secret, file_name, commit_hash)
+
+    def _add_new_secret(self, secret_key: str,
+                        commit_hash: str,
+                        secret: PotentialSecret) -> None:
+        if secret_key not in self.secrets_by_file_value_type:
+            self.secrets_by_file_value_type[secret_key] = []
+        else:
+            all_removed = all(
+                potential_secret.get('removed_commit_hash') for potential_secret in self.secrets_by_file_value_type[secret_key])
+            # Update secret map with the new potential secret
+            if all_removed:
+                self.secrets_by_file_value_type[secret_key][0].update({'potential_secret': secret, 'removed_commit_hash': ''})
+                return
+        self.secrets_by_file_value_type[secret_key].append(
+            {'added_commit_hash': commit_hash, 'removed_commit_hash': '', 'potential_secret': secret})
+
+    def _update_removed_secret(self, secret_key: str,
+                               secret: PotentialSecret,
+                               file_name: str,
+                               commit_hash: str) -> None:
+        # Try to find the corresponding added secret in the gitz t history secret map
+        try:
+            for secret_in_file in self.secrets_by_file_value_type[secret_key]:
                 if secret_in_file['potential_secret'].is_added:
                     secret_in_file['removed_commit_hash'] = commit_hash
                     secret_in_file['potential_secret'] = secret
                     break
-    except KeyError:
-        logging.error(f"No added secret commit found for secret in file {file_name}.")
+        except KeyError:
+            logging.error(f"No added secret commit found for secret in file {file_name}.")
+
+    def handle_renamed_file(self, rename_from: str,
+                            rename_to: str,
+                            commit_hash: str) -> None:
+        temp_map: Dict[str, List[EnrichedPotentialSecret]] = {}
```

Comment:
I'd change this name to something reflecting that we accumulate new secrets here